src/main/java/com/spotify/docker/client/ImageRef.java

```diff
@@ -83,4 +93,23 @@ public String toString() {
         .add("tag", tag)
         .toString();
   }
+
+  /** Return registry server address given first part of image. */
+  @VisibleForTesting
+  static String parseRegistryUrl(final String url) {
+    if (url.equals("docker.io") || url.equals("index.docker.io")) {
+      return DEFAULT_REGISTRY_URL;
+    } else if (url.matches("(^|(\\w+)\\.)gcr\\.io$") || url.equals("gcr.kubernetes.io")) {
+      // GCR uses https
+      return "https://" + url;
+    } else if (url.equals("quay.io")) {
+      // Quay doesn't use any protocol in credentials file
+      return "quay.io";
+    } else if (!url.contains("http://") && !url.contains("https://")) {
+      // Assume http
+      return "http://" + url;
```

Comment:
Docker will interpret a "registry URL" like `example.com/foo/bar` as a registry that uses `https` and will only pull over `http` if specially configured to do so: https://docs.docker.com/engine/reference/commandline/pull/#pull-from-a-different-registry